pyaccel/lifetime.py

```diff
@@ -28,40 +30,140 @@ class Lifetime:
+        self._accepen_neg = accepn
         self._curr_per_bun = 100/864  # [mA]
         self._avg_pressure = 1e-9  # [mbar]
-        self._coupling = 0.03
         self._atomic_number = 7
         self._temperature = 300  # [K]
-        self._taux = self._tauy = self._taue = None
-        self._emit0 = self._espread0 = self._bunlen = None
+        self._tau1 = self._tau2 = self._tau3 = None
+        self._emit1 = self._emit2 = self._espread0 = self._bunlen = None
         self._accepx = self._accepy = self._accepen = None
 
+    @property
+    def type_eqparams_str(self):
+        """."""
+        return Lifetime.EQPARAMS._fields[self._type_eqparams]
+
+    @property
+    def type_eqparams(self):
+        """."""
+        return self._type_eqparams
+
+    @type_eqparams.setter
+    def type_eqparams(self, value):
+        if value is None:
+            return
+        if isinstance(value, str):
+            self._type_eqparams = int(value in Lifetime.EQPARAMS._fields[1])
+        elif int(value) in Lifetime.EQPARAMS:
+            self._type_eqparams = int(value)
+
+    @property
+    def type_optics_str(self):
+        """."""
+        return Lifetime.OPTICS._fields[self._type_optics]
+
+    @property
+    def type_optics(self):
+        """."""
+        return self._type_optics
+
+    @type_optics.setter
+    def type_optics(self, value):
+        if value is None:
+            return
+        if isinstance(value, str):
+            self._type_optics = int(value in Lifetime.OPTICS._fields[1])
+        elif int(value) in Lifetime.OPTICS:
+            self._type_optics = int(value)
+
+    @property
+    def touschek_model_str(self):
+        """."""
+        return Lifetime.TOUSCHEKMODEL._fields[self._touschek_model]
+
+    @property
+    def touschek_model(self):
+        """."""
+        return self._touschek_model
+
+    @touschek_model.setter
+    def touschek_model(self, value):
+        if value is None:
+            return
+        if isinstance(value, str):
+            self._touschek_model = int(
+                value in Lifetime.TOUSCHEKMODEL._fields[1])
+        elif int(value) in Lifetime.TOUSCHEKMODEL:
+            self._touschek_model = int(value)
+
     @property
     def accelerator(self):
         """."""
         return self._acc
 
     @accelerator.setter
     def accelerator(self, val):
-        self._eqpar = _optics.EqParamsFromRadIntegrals(val)
-        res = _optics.calc_transverse_acceptance(val, self._eqpar.twiss)
+        self._eqpar = self._eqparams_func(val)
+        self._optics_data, *_ = self._optics_func(val, indices='closed')
+        _twiss = self._optics_data
+        if self.type_optics != self.OPTICS.Twiss:
+            _twiss, *_ = _optics.calc_twiss(val, indices='closed')
+        res = _optics.calc_transverse_acceptance(val, _twiss)
         self._accepx_nom = _np.min(res[0])
         self._accepy_nom = _np.min(res[1])
+        accepn, accepp = _optics.calc_tousheck_energy_acceptance(val)
```

Comment:
This setter will be slow. Do you realy think it is better to calculate the touscheck acceptance here?
Why not let the user set it?
I think we could change the function `calc_touscheck_energy_acceptance` to have a boolean flag indicating whether it should consider the RF energy acceptance automatically. The way it is today we always have to take the minimum between the transverse dynamics acceptance and the RF acceptance outside the function.

Reply:
Ok I removed  the calculation of touschek_energy_acceptance in the class. Good point, we could discuss this change in `calc_touscheck_energy_acceptance` later in another PR in pyaccel.